src/btree/bt_walk.c

```diff
@@ -65,14 +65,25 @@ __ref_index_slot(WT_SESSION_IMPL *session,
 				goto found;
 			}
 		}
+		/*
+		 * We failed to get the page index and slot reference,
+		 * yield before retrying, and if we've yielded enough times,
+		 * start sleeping so we don't burn CPU to no purpose.
+		 */
+		if (++yield_count < WT_THOUSAND) {
+			__wt_yield();
+			continue;
+		}
 
+		sleep_count = WT_MIN(sleep_count + WT_THOUSAND, 10000);
```

Comment:
maybe 10000 should be 10 * WT_THOUSAND, it is a little easier to read than counting zeroes. But I do not feel strongly about it.